### PR TITLE
Fix failing tests, remove some rails only code and deprecations

### PR DESCRIPTION
**Pull request by javierg**
Since Travis dropped support for open source projects, tests were being neglected, so they actually weren't passing, We will add this project in travis to our org in order be able to run tests. Also removed deprecations from ruby 2.7 / 3.0

### CODE REVIEW

**Comment by javierg** [comment]
Please reviewer, run tests locally before approving the code

**Comment by alejandrodevs** [comment]
@javierg I'm still getting an error in ruby 3.0.0
undefined method `DelegateClass' for Dolly:Module

I think that method is deprecated but we can use SimpleDelegator.

```ruby
require 'delegate'

module Dolly
  class Collection < SimpleDelegator
```

**Comment by javierg** [comment]
@alejandrodevs  I am still trying to setup Travis, so no test on other ruby version but the one defined in the .ruby-version file, I am gonna do a test on 3.0, but we still need to support 2.6, so will check what options are there, thanks

**Review by alejandrodevs — APPROVED** [review]
:shipit: 
@javierg Tests are running correctly in 2.6.5.

**Comment by javierg** [comment]
@alejandrodevs  can you try on 3.0 too please.

**Comment by alejandrodevs** [comment]
@javierg Now it is working on 3.0 👏🏼

File: lib/dolly/collection.rb [hunk 1]
```diff
@@ -21,7 +23,7 @@ def single?
     private
 
     def collect_docs
-      lambda do |row|
+      proc do |row|
```
Comment (javierg) [inline lib/dolly/collection.rb:26]: On ruby 3.0 we cant call lambdas as proc arguments `any_method(&methd_returning_proc)`  wotn work if the method returns a `lambda`

**Pull request merged**